src/d1to2fix/symbolsearch.d

```diff
@@ -63,7 +72,15 @@ public void initializeModuleCache(string[] paths)
                 || symbol.kind == CompletionKind.interfaceName
                 || symbol.kind == CompletionKind.className)
             {
-                collectNames((*symbol)[]);
+                try
+                {
+                    collectNames((*symbol)[]);
+                }
+                catch (SymbolSearchRecursionError e)
+                {
+                    e.add(symbol.name ~ "@" ~ symbol.symbolFile);
+                    throw e;
+                }
```

Comment:
Move the check for stack length outside of the error class, then:

```
if (depth > 20)
  collectNames((*symbol)[]);
else
{
                try
                {
                    collectNames((*symbol)[]);
                }
                catch (SymbolSearchRecursionError e)
                {
                    e.add(symbol.name ~ "@" ~ symbol.symbolFile);
                    throw e;
                }
}
```

A bit ugly but should do the trick